i18n-helpers/src/lib.rs

```diff
@@ -569,12 +569,22 @@ fn check_code_block_token_count(group: &[(usize, Event)]) -> usize {
     let events = group.iter().map(|(_, event)| event);
     let mut in_codeblock = false;
     let mut max_token_count = 0;
+
+    // token_count should be taken over Text events
+    // because a continuous text may be splitted to some Text events.
+    let mut token_count = 0;
+
     for event in events {
         match event {
-            Event::Start(Tag::CodeBlock(_)) => in_codeblock = true,
-            Event::End(Tag::CodeBlock(_)) => in_codeblock = false,
+            Event::Start(Tag::CodeBlock(_)) => {
+                in_codeblock = true;
+                token_count = 0;
+            }
+            Event::End(Tag::CodeBlock(_)) => {
+                in_codeblock = false;
+                token_count = 0;
+            }
             Event::Text(x) if in_codeblock => {
```

Comment:
I think this `Event::Text` arm will be easier to understand if `max_token_count` is updated only in a single place:

```
            Event::Text(x) if in_codeblock => {
                for c in x.chars() {
                    if c == '`' {
                        token_count += 1;
                        max_token_count = std::cmp::max(max_token_count, token_count);
                    } else {
                        token_count = 0;
                    }
                }
            }
```